lib/temporal/activity/task_processor.rb

```diff
@@ -26,16 +29,17 @@ def process
           raise ActivityNotRegistered, 'Activity is not registered with this worker'
         end
 
-        metadata = Metadata.generate(Metadata::ACTIVITY_TYPE, task, namespace)
         context = Activity::Context.new(client, metadata)
 
-        result = middleware_chain.invoke(metadata) do
+        result = middleware_chain.invoke(metadata, task.to_h) do
           activity_class.execute_in_context(context, parse_payload(task.input))
         end
 
         # Do not complete asynchronous activities, these should be completed manually
         respond_completed(result) unless context.async?
       rescue StandardError, ScriptError => error
+        Temporal::ErrorHandler.handle(error, metadata: metadata.to_h)
```

Comment:
Btw, I see where you coming from with `.to_h`, but compared to logs it's actually that bad to pass the whole metadata object here. Any reasons you decided not to do it?

Reply:
Is would allow devs to distinguish between different metadata types

Reply:
I didn't want application developers to have to know the metadata class internals. If we send a hash then there is no API concerns. I don't feel strongly about it though if you'd prefer to send the objects I can `.to_h` in my application's handler.